samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -2,3 +2,41 @@
 Use Terraform plan to link resources together
 e.g. linking layers to functions
 """
+
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+
+@dataclass
+class Value:
+    constant_value: Any
+    references: List[str]
+
+
+@dataclass
+class Module:
+    # full path to the module, including parent modules
+    full_address: str
+    parent_module: Optional["Module"]
+    variables: Dict[str, Value]
+    resources: Dict[str, "Resource"]
+    children_modules: Dict[str, "Module"]
+    outputs: Dict[str, Value]
+
+    # current module's + all child modules' resources
+    def get_all_resources(self) -> List["Resource"]:
+        # TODO
+        pass
+
+
+@dataclass
+class Resource:
```

Comment:
nit: We can move this class declaration above `Module` since it is used inside of `Module`. Then we can reference `Resources` in the type hints without the `""`.

Reply:
`Resource` also references a `Module`, so it's kinda the same either way 😅

Reply:
I must've missed that, my bad. We can probably remove the quotes around Module defined here then.